jobsdb/prebackup/prebackup_test.go

```diff
@@ -0,0 +1,193 @@
+	if err != nil {
+		log.Printf("Could not start postgres: %s", err)
+		return -1
+	}
+
+	fmt.Println("DB_DSN:", postgresql.DB_DSN)
+	return m.Run()
+}
+
+func withBlockOnHold(f func() int) int {
+	var hold bool
+	flag.BoolVar(&hold, "hold", false, "hold environment clean-up after test execution until Ctrl+C is provided")
+	flag.Parse()
+
+	res := f()
+
+	if !hold {
+		return res
+	}
+	fmt.Println("Test on hold, before cleanup")
+	fmt.Println("Press Ctrl+C to exit")
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	<-c
+	return res
+}
+
+func createTables(t *testing.T, txn *sql.Tx, jobsTable, jobStatusTable string) {
+	sqlStatement := fmt.Sprintf(`CREATE TABLE "%s" (
+		job_id BIGSERIAL PRIMARY KEY,
+		workspace_id TEXT NOT NULL DEFAULT '',
+		uuid UUID NOT NULL,
+		user_id TEXT NOT NULL,
+		parameters JSONB NOT NULL,
+		custom_val VARCHAR(64) NOT NULL,
+		event_payload JSONB NOT NULL,
+		event_count INTEGER NOT NULL DEFAULT 1,
+		created_at TIMESTAMP NOT NULL DEFAULT NOW(),
+		expire_at TIMESTAMP NOT NULL DEFAULT NOW());`, jobsTable)
+	_, err := txn.Exec(sqlStatement)
+	if err != nil {
+		t.Fatalf("Failed to create jobs table: %s", err)
+	}
+
+	sqlStatement = fmt.Sprintf(`CREATE TABLE "%s" (
+			id BIGSERIAL,
+			job_id BIGINT REFERENCES "%s"(job_id),
+			job_state VARCHAR(64),
+			attempt SMALLINT,
+			exec_time TIMESTAMP,
+			retry_time TIMESTAMP,
+			error_code VARCHAR(32),
+			error_response JSONB DEFAULT '{}'::JSONB,
+			parameters JSONB DEFAULT '{}'::JSONB,
+			PRIMARY KEY (job_id, job_state, id));`, jobStatusTable, jobsTable)
+	_, err = txn.Exec(sqlStatement)
+	if err != nil {
+		t.Fatalf("Failed to create job status table: %s", err)
+	}
+}
+
+func addJob(t *testing.T, txn *sql.Tx, jobsTable, jobStatusTable, sourceId, state string) {
+	sqlStatement := fmt.Sprintf(`INSERT INTO "%s" (workspace_id, uuid, user_id, parameters, custom_val, event_payload, event_count) VALUES(
+		'workspace_id',
+		'a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11',
+		'uder_id',
+		'{ "source_id": "%s" }',
+		'custom_val',
+		'{}',
+		1);`, jobsTable, sourceId)
+	_, err := txn.Exec(sqlStatement)
+	if err != nil {
+		t.Fatalf("failed to insert job entry: %s", err)
+	}
+	sqlStatement = fmt.Sprintf(`INSERT INTO "%s" (job_id, job_state, attempt) VALUES(
+		(SELECT max(job_id) FROM "%s") ,
+		'%s',
+		1);`, jobStatusTable, jobsTable, state)
+	_, err = txn.Exec(sqlStatement)
+	if err != nil {
+		t.Fatalf("failed to insert job status entry: %s", err)
+	}
+
+}
+
+func countRows(t *testing.T, txn *sql.Tx, tableName string) int {
+	rows, err := txn.Query(fmt.Sprintf("SELECT COUNT(*) as count FROM  %s", tableName))
+	if err != nil {
+		t.Fatalf("failed to count rows for table %s: %s", tableName, err)
+	}
+	return checkCount(t, rows)
+}
+
+func checkCount(t *testing.T, rows *sql.Rows) (count int) {
+	for rows.Next() {
+		err := rows.Scan(&count)
+		if err != nil {
+			t.Fatalf("failed to count rows: %s", err)
+		}
+	}
+	return count
+}
```

Comment:
[optional]

```suggestion
func requireRowsCount(t *testing.T, txn *sql.Tx, tableName string, expectedCount int) {
	t.Helper()
	rows := txn.QueryRow(fmt.Sprintf("SELECT COUNT(*) as count FROM  %s", tableName))

	var count int
	require.NoError(t, rows.Scan(&count))
	require.EqualValues(t, expectedCount, count)
}
```

Given that you're passing `t` already we could have it as a `t.Helper()` and get rid of `checkCount` thanks to `txn.QueryRow`.

Reply:
thanks!